homeassistant/components/iaqualink/binary_sensor.py

```diff
@@ -0,0 +1,54 @@
+"""Support for Aqualink temperature sensors."""
+import logging
+
+from iaqualink import AqualinkBinarySensor
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
+    DEVICE_CLASS_COLD,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import AqualinkEntity
+from .const import DOMAIN as AQUALINK_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up discovered binary sensors."""
+    devs = []
+    for dev in hass.data[AQUALINK_DOMAIN][DOMAIN]:
+        devs.append(HassAqualinkBinarySensor(dev))
+    async_add_entities(devs, True)
+
+
+class HassAqualinkBinarySensor(BinarySensorDevice, AqualinkEntity):
+    """Representation of a binary sensor."""
+
+    def __init__(self, dev: AqualinkBinarySensor):
```

Comment:
We can remove this since we don't do anything else than calling the parent init method.

Reply:
Ok, I'll have to move the base classes around to ensure MRO is correct then.

Reply:
To be future safe that would be good. None of these platform base entity classes have init methods.